runners/spark/src/main/java/org/apache/beam/runners/spark/io/MicrobatchSource.java

```diff
@@ -201,56 +201,66 @@ public int hashCode() {
    */
```

Comment:
@dhalperi , while I was not able to fully remove the duplicate code, I did some cleanups that simplify things a bit.